### ISSUE TITLE
New Lint: blacklisted_method

### ISSUE THREAD

**Issue by ilknarf** [rust-lang/rust-clippy#6073]
### What it does
~~Blacklist Method~~ Disallowed Method (see https://github.com/rust-lang/rust-clippy/pull/5896)
*What does this lint do?*
Disallows certain methods, as defined in `clippy.toml`
### Categories (optional)

- Kind: Nursery

Checks for method calls of specified methods
For example:
- `SomeStruct::bad_method`
-  `SomeTrait::bad_method`

### Drawbacks
This might take some significant overhead to implement (regex or some manual string parsing), and the main purpose would probably be to patch in temporary lints, so it might not be super useful.
### Example
```
# clippy.toml

disallowed-methods = ["Foo::bar", "Baz::bar"]
```

```rust
...
let f = Foo::default();
f.bar();
...
```
results in a lint

**Comment by flip1995** [comment]
Should only the usage be blacklisted or also the definition?

**Comment by ilknarf** [comment]
We were mostly looking at usage.

**Comment by mikerite** [comment]
Apparently we don't use the term "blacklist" anymore - #5896 .  Maybe choose something else if you don't want someone to come along and rename it.

**Comment by ilknarf** [comment]
Ok I vibe with that. Would `banned_method` be appropriate nomenclature for the future?

**Comment by mikerite** [comment]
The other PR used the term 'disallowed' so maybe go with `disallowed_method`. I probably not the right person to answer this.

**Comment by ilknarf** [comment]
Many thanks! I will go with that, then.
